src/react-queries/queries/thornode.ts

```diff
@@ -1,81 +1,106 @@
 import axios from 'axios'
 
 import { getConfig } from '@/config'
 import type { ThorchainBlock, ThorchainMimir } from '@/lib/utils/thorchain/types'
 
-const thornodeUrl = getConfig().VITE_THORCHAIN_NODE_URL
+export const createThornodeQueries = <T extends string>(key: T, baseUrl: string) => {
+  return createQueryKeys(key, {
+    poolData: (assetId: AssetId | undefined) => ({
+      queryKey: [`${key}PoolData`, assetId] as const,
+      queryFn: async () => {
+        if (!assetId) throw new Error('assetId is required')
+
+        const poolAssetId = assetIdToThorPoolAssetId({ assetId })
+        const { data } = await axios.get<ThornodePoolResponse>(`${baseUrl}/pool/${poolAssetId}`)
 
-export const fetchThorchainMimir = async () => {
-  const { data } = await axios.get<ThorchainMimir>(`${thornodeUrl}/thorchain/mimir`)
-  return data
+        return data
+      },
+    }),
+    poolsData: () => ({
+      queryKey: [`${key}PoolsData`],
+      queryFn: async () => {
+        const { data } = await axios.get<ThornodePoolResponse[]>(`${baseUrl}/pools`)
+        return data
+      },
+    }),
+    mimir: () => {
+      return {
+        queryKey: [`${key}Mimir`] as const,
+        queryFn: async () => {
+          const { data } = await axios.get<ThorchainMimir>(`${baseUrl}/mimir`)
+          return data
+        },
+      }
+    },
+    block: () => {
+      return {
+        queryKey: [`${key}BlockHeight`],
+        queryFn: async () => {
+          const { data } = await axios.get<ThorchainBlock>(`${baseUrl}/block`)
+          return data
+        },
+      }
+    },
+    inboundAddresses: () => {
+      return {
+        queryKey: [`${key}InboundAddress`],
+        queryFn: async () => {
+          const { data } = await axios.get<InboundAddressResponse[]>(`${baseUrl}/inbound_addresses`)
+          return data.filter(v => !v.halted)
+        },
+      }
+    },
+  })
 }
 
-// Feature-agnostic, abstracts away THORNode endpoints
-export const thornode = createQueryKeys('thornode', {
-  poolData: (assetId: AssetId | undefined) => ({
-    queryKey: ['thornodePoolData', assetId],
-    queryFn: async () => {
-      if (!assetId) throw new Error('assetId is required')
+export const thornode = createThornodeQueries(
+  'thornode',
+  `${getConfig().VITE_THORCHAIN_NODE_URL}/thorchain`,
+)
 
-      const poolAssetId = assetIdToPoolAssetId({ assetId })
-      const { data } = await axios.get<ThornodePoolResponse>(
-        `${thornodeUrl}/thorchain/pool/${poolAssetId}`,
-      )
+export const mayanode = createThornodeQueries(
+  'mayanode',
+  `${getConfig().VITE_MAYACHAIN_NODE_URL}/mayachain`,
+)
 
-      return data
-    },
-  }),
-  poolsData: () => ({
-    queryKey: ['thornodePoolsData'],
-    queryFn: async () => {
-      const poolResponse = await thorService.get<ThornodePoolResponse[]>(
-        `${thornodeUrl}/thorchain/pools`,
-      )
+export const getInboundAddressesQuery = (chainId: ChainId) => {
```

Comment:
This was the most ergonomic I could manage to allow for easiest possible conditional calls against thorchain or mayachain given a chainId. Open to feedback though.

Reply:
^ note the comment above refers to `createQueryKeys()` bits - as far as this is concerned, this is perfect IMO!